app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponseDao.java

```diff
@@ -105,34 +103,41 @@ public SurveyResponse appendSurveyAnswers(SurveyResponse response, List<SurveyAn
     }
     
     @Override
-    public void deleteSurveyResponse(SurveyResponse response) {
-        responseMapper.delete(response);
+    public void deleteSurveyResponses(String healthCode) {
+        DynamoSurveyResponse hashKey = new DynamoSurveyResponse();
+        hashKey.setHealthCode(healthCode);
+        
+        DynamoDBQueryExpression<DynamoSurveyResponse> query = new DynamoDBQueryExpression<DynamoSurveyResponse>();
+        query.setHashKeyValues(hashKey);
+        PaginatedQueryList<DynamoSurveyResponse> results = mapper.query(DynamoSurveyResponse.class, query);
+        
+        List<DynamoSurveyResponse> responsesToDelete = Lists.newArrayList();
+        responsesToDelete.addAll(results);
```

Comment:
What you say makes sense, which is why I've tried it multiple times, and it won't delete. You have to copy to a list before the delete will work.